Sources/CXUtility/Locking.swift

```diff
@@ -0,0 +1,272 @@
+import CXLibc
+
+public protocol Locking {
+    
+    func lock()
+    
+    func tryLock() -> Bool
+    
+    func unlock()
+}
+
+extension Locking {
+    
+    public func withLock<T>(_ body: () throws -> T) rethrows -> T {
+        self.lock(); defer { self.unlock() }
+        return try body()
+    }
+    
+    public func withLockGet<T>(_ body: @autoclosure () throws -> T) rethrows -> T {
+        self.lock(); defer { self.unlock() }
+        return try body()
+    }
+}
+
+// MARK: - Lock
+
+public struct Lock: Locking {
```

Comment:
Should use `class` here.
Sorry I didn't notice this at first.
@ddddxxx

Reply:
I've forgotten the original post I read before, but this thread also explains why we should use `class` to wrap a mutex.
https://forums.swift.org/t/atomic-property-wrapper-for-standard-library/30468/25

Reply:
I intentionally use struct to avoid ARC traffic, [like stdlib did](https://github.com/apple/swift/blob/master/stdlib/public/Darwin/Foundation/Publishers%2BLocking.swift). The `Lock` instance doesn't need a stable address. It simply wrap a `UnsafeMutablePointer<os_unfair_lock_s>`. The `os_unfair_lock_s` pointee has a stable address.

Reply:
Yes, you are right!
Glad to learn that~ Thanks!

Reply:
Actually the main goal of this pr is to make `Lock` struct 😆.